### ISSUE TITLE
loop count error

### ISSUE THREAD

**Issue by natman44** [koral--/android-gif-drawable#204]
1、GifDrawable.setLoopCount(5)
2、GifDrawable.getLoopCount()
3、GifDrawable.getCurrentLoop()

the getCurrentLoop() is error.
the log as follows

09-10 16:25:37.006: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=0
09-10 16:25:47.676: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=1
09-10 16:25:58.026: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=2
09-10 16:26:09.966: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=4
09-10 16:26:20.236: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=4
09-10 16:27:32.166: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=1
09-10 16:27:42.836: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=1
09-10 16:27:53.316: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=2
09-10 16:28:03.786: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=4
09-10 16:28:14.136: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=5
09-10 16:28:59.206: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=1
09-10 16:29:11.206: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=2
09-10 16:29:22.026: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=2
09-10 16:29:32.396: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=3
09-10 16:29:42.986: E/strongtv(26270): tv.huan.app.business.view.ScheduleTips$1$1$1.onAnimationCompleted()::###################### loopCount=5 curLoop=4

**Comment by koral--** [comment]
It is a bug that `GifDrawable.getCurrentLoop()` may return value equal to number of loop count. That value is used internally to indicate that all the loops has completed but should not be returned by public API.

Issue with `GifDrawable.getCurrentLoop()` called in `AnimationListener.onAnimationCompleted()` returning not monotonically increasing values is partially caused by design, but also needs fixing.

TL;DR
By default frame rendering happens in background thread and when last frame is rendered `AnimationListener.onAnimationCompleted()` is scheduled to be fired in UI thread with delay equal to the last frame duration (with respect to speed factor and time lost on decoding).
When you are calling `GifDrawable.getLoopCount()` it reads up-to-date value which may be higher than index of the last completed loop because rendering thread may manage to proceed to the next frame before `onAnimationCompleted()` is actually invoked.

**Comment by koral--** [comment]
Fixed, value returned by `GifDrawable.getCurrentLoop()` should not exceed bounds and `AnimationListener.onAnimationCompleted()` has now parameter `loopNumber`.
